File: sistemaacademicojcapi/app/Models/PersonaRol.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasOne;

class PersonaRol extends Model
{
    protected $table = 'persona_rol';
    protected $primaryKey = 'id_persona_rol';
    public $timestamps = false;

    protected $fillable = [
        'roles_id_rol',
        'personas_id_persona',
    ];

    // Relaciones

    public function persona(): BelongsTo
    {
        return $this->belongsTo(Persona::class, 'personas_id_persona', 'id_persona');
    }

    public function rol(): BelongsTo
    {
        return $this->belongsTo(Rol::class, 'roles_id_rol', 'id_rol');
    }

    public function user(): HasOne
    {
        return $this->hasOne(User::class, 'persona_rol_id_persona_rol', 'id_persona_rol');
    }
}

File: sisitemacademicoappmovil/SistemaAcademicoJCT/app/src/main/java/com/jsq/sistemaacademicojct/model/PersonaRol.java
package com.jsq.sistemaacademicojct.model;

public class PersonaRol {
    private int id_persona_rol;
    private int roles_id_rol;
    private int personas_id_persona;
    private Rol rol;
    private Persona persona;

    public int getId_persona_rol() {
        return id_persona_rol;
    }

    public Rol getRol() {
        return rol;
    }

    public Persona getPersona() {
        return persona;
    }
}
